diff --git a/components/fxa-client/android/build.gradle b/components/fxa-client/android/build.gradle
index 88709f675..165ed22f2 100644
--- a/components/fxa-client/android/build.gradle
+++ b/components/fxa-client/android/build.gradle
@@ -2,14 +2,7 @@ buildscript {
     // in mozilla-central there's no build.gradle suitable for the maven/glean config, so each component gets it.
     if (gradle.hasProperty("mozconfig")) {
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
 
         dependencies {
diff --git a/components/logins/android/build.gradle b/components/logins/android/build.gradle
index 748c8fc02..aaa73f6e4 100644
--- a/components/logins/android/build.gradle
+++ b/components/logins/android/build.gradle
@@ -2,14 +2,7 @@ buildscript {
     // in mozilla-central there's no build.gradle suitable for the maven/glean config, so each component gets it.
     if (gradle.hasProperty("mozconfig")) {
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
 
         dependencies {
diff --git a/components/nimbus/android/build.gradle b/components/nimbus/android/build.gradle
index 74b3aec47..14cc9bd64 100644
--- a/components/nimbus/android/build.gradle
+++ b/components/nimbus/android/build.gradle
@@ -2,14 +2,7 @@ buildscript {
     // in mozilla-central there's no build.gradle suitable for the maven/glean config, so each component gets it.
     if (gradle.hasProperty("mozconfig")) {
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
 
         dependencies {
diff --git a/components/places/android/build.gradle b/components/places/android/build.gradle
index be15dd740..47afad31d 100644
--- a/components/places/android/build.gradle
+++ b/components/places/android/build.gradle
@@ -2,14 +2,7 @@ buildscript {
     // in mozilla-central there's no build.gradle suitable for the maven/glean config, so each component gets it.
     if (gradle.hasProperty("mozconfig")) {
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
 
         dependencies {
diff --git a/components/sync_manager/android/build.gradle b/components/sync_manager/android/build.gradle
index 48d962f38..6bff91c91 100644
--- a/components/sync_manager/android/build.gradle
+++ b/components/sync_manager/android/build.gradle
@@ -2,14 +2,7 @@ buildscript {
     // in mozilla-central there's no build.gradle suitable for the maven/glean config, so each component gets it.
     if (gradle.hasProperty("mozconfig")) {
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
 
         dependencies {
diff --git a/settings.gradle b/settings.gradle
index 34feca4cd..1c083df84 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -31,14 +31,7 @@ buildscript {
         gradle.ext.mozconfig = gradle.root.mozconfig
 
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
     }
 }
diff --git a/tools/nimbus-gradle-plugin/settings.gradle b/tools/nimbus-gradle-plugin/settings.gradle
index 7d907f057..8679108f7 100644
--- a/tools/nimbus-gradle-plugin/settings.gradle
+++ b/tools/nimbus-gradle-plugin/settings.gradle
@@ -21,14 +21,7 @@ buildscript {
         gradle.ext.mozconfig = gradle.root.mozconfig
 
         repositories {
-            gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-                maven {
-                    url = repository
-                    if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                        allowInsecureProtocol = true
-                    }
-                }
-            }
+            mavenLocal()
         }
     }
 }
