diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt
index 93b95ae48855..a2d739c32900 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt
@@ -268,7 +268,7 @@ class Components(private val context: Context) {
     val publicSuffixList by lazyMonitored { PublicSuffixList(context) }
     val clipboardHandler by lazyMonitored { ClipboardHandler(context) }
     val performance by lazyMonitored { PerformanceComponent() }
-    val push by lazyMonitored { Push(context, analytics.crashReporter) }
+    val push by lazyMonitored { Push(context) }
     val wifiConnectionMonitor by lazyMonitored { WifiConnectionMonitor(context as Application) }
 
     val strictMode by lazyMonitored {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt
index 1caab673268c..e23091791ea9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt
@@ -31,7 +31,7 @@ class Push(val context: Context, val strictMode: StrictModeManager = context.com
     private val logger = Logger("Push")
     val feature: Pusher?
         get() = strictMode.allowViolation(StrictMode::allowThreadDiskReads) {
-            if (context.settings().ironfox.useUnifiedPush) {
+            if (context.components.settings.useUnifiedPush) {
                 unifiedPushFeature.also { feature ->
                     if (feature.isAvailable(context)) {
                         logger.info("Using UnifiedPush")
@@ -50,7 +50,7 @@ class Push(val context: Context, val strictMode: StrictModeManager = context.com
                             notificationId = notificationId,
                             notification = notification
                         )
-                        context.settings().ironfox.useUnifiedPush = false
+                        context.components.settings.useUnifiedPush = false
                     }
                 }
             } else {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index bcf30637b845..24ab8b93f40b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -176,6 +176,12 @@ class SecretSettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFra
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
 
+        requirePreference<SwitchPreferenceCompat>(R.string.pref_key_enable_unifiedpush).apply {
+            isVisible = true
+            isChecked = settings.enableUnifiedPush
+            onPreferenceChangeListener = SharedPreferenceUpdater()
+        }
+
         requirePreference<SwitchPreferenceCompat>(R.string.pref_key_enable_address_sync).apply {
             isVisible = Config.channel.isNightlyOrDebug
             isChecked = settings.isAddressSyncEnabled
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index 0730e7a96bc8..230bf096a649 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -478,6 +478,48 @@ class SettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFragment
                 null
             }
 
+            // Only displayed when UnifiedPush is enabled
+            resources.getString(R.string.pref_key_use_unifiedpush) -> {
+                val context = requireActivity()
+                context.components.settings.apply { useUnifiedPush = !useUnifiedPush }
+                val alert = MaterialAlertDialogBuilder(context).apply {
+                    setTitle(context.getString(R.string.preferences_unifiedpush))
+                    setMessage(context.getString(R.string.quit_application))
+                    setNegativeButton(android.R.string.cancel) { dialog: DialogInterface, _ ->
+                        dialog.cancel()
+                    }
+
+                    setPositiveButton(android.R.string.ok) { _, _ ->
+                        Toast.makeText(
+                            context,
+                            getString(R.string.toast_change_unifiedpush_done),
+                            Toast.LENGTH_LONG,
+                        ).show()
+
+                        Handler(Looper.getMainLooper()).postDelayed(
+                            {
+                                exitProcess(0)
+                            },
+                            FXA_SYNC_OVERRIDE_EXIT_DELAY,
+                        )
+                    }
+                    create().withCenterAlignedButtons()
+                }
+                if (context.components.settings.useUnifiedPush) {
+                    requireComponents.push.switchToUnifiedPush(context) { success ->
+                        if (!success) {
+                            context.components.settings.useUnifiedPush = false
+                        } else {
+                            alert.show()
+                        }
+                    }
+                } else {
+                    requireComponents.push.switchToAutoPush(context)
+                    alert.show()
+                }
+                null
+            }
+
             // Only displayed when secret settings are enabled
             resources.getString(R.string.pref_key_override_amo_collection) -> {
                 val context = requireContext()
@@ -614,6 +656,12 @@ class SettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFragment
             findPreference<Preference>(getPreferenceKey(R.string.pref_key_start_profiler))
 
         with(settings) {
+            findPreference<SwitchPreferenceCompat>(
+                getPreferenceKey(R.string.pref_key_use_unifiedpush),
+            )?.apply {
+                isVisible = enableUnifiedPush
+                isChecked = useUnifiedPush
+            }
             findPreference<Preference>(
                 getPreferenceKey(R.string.pref_key_nimbus_experiments),
             )?.isVisible = showSecretDebugMenuThisSession
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index d4bf7f210f9d..a7e282e748d3 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2296,6 +2296,22 @@ class Settings(
         }
     }
 
+    /**
+     * Indicates if UnifiedPush can be used
+     */
+    var enableUnifiedPush by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_enable_unifiedpush),
+        default = true,
+    )
+
+    /**
+     * Indicates if UnifiedPush is used
+     */
+    var useUnifiedPush by booleanPreference(
+        key = appContext.getPreferenceKey(R.string.pref_key_use_unifiedpush),
+        default = false,
+    )
+
     /**
      * Get the current mode for cookie banner handling
      */
diff --git a/mobile/android/fenix/app/src/main/res/values/preference_keys.xml b/mobile/android/fenix/app/src/main/res/values/preference_keys.xml
index fe54d857432b..3b5e0da100fc 100644
--- a/mobile/android/fenix/app/src/main/res/values/preference_keys.xml
+++ b/mobile/android/fenix/app/src/main/res/values/preference_keys.xml
@@ -460,6 +460,8 @@
     <string name="pref_key_pocket_sponsored_stories" translatable="false">pref_key_pocket_sponsored_stories</string>
 
     <!-- Secret Info Setting Keys -->
+    <string name="pref_key_enable_unifiedpush" translatable="false">pref_key_enable_unifiedpush</string>
+    <string name="pref_key_use_unifiedpush" translatable="false">pref_key_use_unifiedpush</string>
     <string name="pref_key_start_profiler" translatable="false">pref_key_start_profiler</string>
 
     <!-- Secret Settings -->
diff --git a/mobile/android/fenix/app/src/main/res/values/static_strings.xml b/mobile/android/fenix/app/src/main/res/values/static_strings.xml
index c732f143d510..6acb7ddffbdc 100644
--- a/mobile/android/fenix/app/src/main/res/values/static_strings.xml
+++ b/mobile/android/fenix/app/src/main/res/values/static_strings.xml
@@ -71,6 +71,8 @@
     <string name="preferences_nimbus_experiments_inactive">Inactive Experiments</string>
     <!-- Label for using the nimbus collections preview -->
     <string name="preferences_nimbus_use_preview_collection">Use Nimbus Preview Collection (requires restart)</string>
+    <!-- Label for enabling UnifiedPush -->
+    <string name="preferences_debug_settings_unifiedpush" translatable="false">Enable UnifiedPush</string>
     <!-- Label for custom Glean server URL -->
     <string name="preferences_debug_settings_custom_glean_server_url" translatable="false">Custom Glean server URL (requires restart)</string>
     <!-- Title of preference for sync debugging (only shown in the when the secret debug menu is enabled) -->
diff --git a/mobile/android/fenix/app/src/main/res/values/strings.xml b/mobile/android/fenix/app/src/main/res/values/strings.xml
index 8fa480ab620e..ecf4f10258c1 100644
--- a/mobile/android/fenix/app/src/main/res/values/strings.xml
+++ b/mobile/android/fenix/app/src/main/res/values/strings.xml
@@ -924,6 +924,13 @@
     <!-- Message to indicate users that we are quitting the application to apply the changes -->
     <string name="quit_application">Quitting the application to apply changes…</string>
 
+    <!-- Label for using UnifiedPush. UnifiedPush is a decentralized push notification system, so users can benefit from mobile push notifications without relying on Google Services. Users choose the application (and server) they wish to use to manage webpush for other applications. If enabled, the browser (e.g. Firefox for Android) will receive webpush messages via the chosen app. -->
+    <string name="preferences_unifiedpush">Use UnifiedPush</string>
+    <!--  The application must be restarted for the UnifiedPush setting to take effect. -->
+    <string name="preferences_unifiedpush_summary">Requires restart</string>
+    <!-- Toast shown after confirming the UnifiedPush configuration -->
+    <string name="toast_change_unifiedpush_done">UnifiedPush setting modified. Quitting the application to apply changes…</string>
+
     <!-- Preference for extensions -->
     <string name="preferences_extensions">Extensions</string>
     <!-- Preference for installing a local extension -->
diff --git a/mobile/android/fenix/app/src/main/res/xml/preferences.xml b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
index 2691e3053f3f..3f6834b5f519 100644
--- a/mobile/android/fenix/app/src/main/res/xml/preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
@@ -214,6 +214,13 @@
             app:isPreferenceVisible="false"
             android:title="@string/firefox_labs_title" />
 
+        <androidx.preference.SwitchPreferenceCompat
+            android:defaultValue="false"
+            android:key="@string/pref_key_use_unifiedpush"
+            app:iconSpaceReserved="false"
+            android:title="@string/preferences_unifiedpush"
+            android:summary="@string/preferences_unifiedpush_summary"/>
+
         <androidx.preference.SwitchPreferenceCompat
             android:key="@string/pref_key_leakcanary"
             android:title="@string/preference_leakcanary"
diff --git a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
index 6f2fbf7f975a..71b44cd05a66 100644
--- a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
@@ -178,6 +178,11 @@
       android:key="@string/pref_key_nimbus_use_preview"
       android:title="@string/preferences_nimbus_use_preview_collection"
       app:iconSpaceReserved="false" />
+    <SwitchPreferenceCompat
+        android:defaultValue="false"
+        android:key="@string/pref_key_enable_unifiedpush"
+        android:title="@string/preferences_debug_settings_unifiedpush"
+        app:iconSpaceReserved="false" />
     <SwitchPreferenceCompat
       android:defaultValue="false"
       android:key="@string/pref_key_enable_firefox_labs"
