diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
index 0d72d61e72a8..735a3a7534c3 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
@@ -7,7 +7,6 @@ package org.mozilla.fenix.gecko
 import android.content.Context
 import androidx.annotation.VisibleForTesting
 import mozilla.components.browser.engine.gecko.autofill.GeckoAutocompleteStorageDelegate
-import mozilla.components.browser.engine.gecko.crash.GeckoCrashPullDelegate
 import mozilla.components.browser.engine.gecko.ext.toContentBlockingSetting
 import mozilla.components.concept.engine.EngineSession.TrackingProtectionPolicy
 import mozilla.components.concept.storage.CreditCardsAddressesStorage
@@ -71,15 +70,6 @@ object GeckoProvider {
                 ),
             )
 
-        geckoRuntime.crashPullDelegate =
-            GeckoCrashPullDelegate(
-                dispatcher = { crashIDs ->
-                    context.components.appStore.dispatch(
-                        AppAction.CrashActionWrapper(CrashAction.CheckDeferred(crashIDs.toList()))
-                    )
-                }
-            )
-
         return geckoRuntime
     }
 
@@ -117,7 +107,6 @@ object GeckoProvider {
                 .extensionsProcessEnabled(true)
                 .extensionsWebAPIEnabled(true)
                 .translationsOfferPopup(context.components.settings.offerTranslation)
-                .crashPullNeverShowAgain(context.components.settings.crashPullNeverShowAgain)
                 .setSameDocumentNavigationOverridesLoadType(
                     FxNimbus.features.sameDocumentNavigationOverridesLoadType.value().enabled
                 )
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index d7ccff8d6d36..47bef0aaf2aa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -434,7 +434,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFra
         }
 
         requirePreference<SwitchPreferenceCompat>(R.string.pref_key_crash_pull_never_show_again).apply {
-            isVisible = true
+            isVisible = false
             isChecked = settings.crashPullNeverShowAgain
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 69e9f18a2457..22aa835c2bc9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -706,11 +706,7 @@ class Settings(
                     true,
                 )
 
-    var crashReportChoice by
-        stringPreference(
-            appContext.getPreferenceKey(R.string.pref_key_crash_reporting_choice),
-            default = CrashReportOption.Ask.toString(),
-        )
+    var crashReportChoice = CrashReportOption.Never.toString()
 
     val isRemoteDebuggingEnabled by
         booleanPreference(
@@ -1162,11 +1158,7 @@ class Settings(
         )
 
     /** Indicates if the user denies to ever see again the Remote Settings crash pull UI. */
-    var crashPullNeverShowAgain: Boolean by
-        booleanPreference(
-            appContext.getPreferenceKey(R.string.pref_key_crash_pull_never_show_again),
-            default = false,
-        )
+    var crashPullNeverShowAgain = false
 
     @VisibleForTesting internal fun timeNowInMillis(): Long = currentTimeMillis()
 
