diff -r 7a43394d2f4f build.gradle
--- a/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     ext {
@@ -134,14 +129,9 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     // Use the semanticdb-javac and semanticdb-kotlinc plugins to generate semanticdb files for Searchfox
diff -r 7a43394d2f4f mobile/android/android-components/build.gradle
--- a/mobile/android/android-components/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -9,14 +9,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
@@ -40,14 +35,9 @@
 
 allprojects {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 
         maven {
             url "${gradle.mozconfig.topobjdir}/gradle/maven"
diff -r 7a43394d2f4f mobile/android/android-components/components/browser/engine-gecko/build.gradle
--- a/mobile/android/android-components/components/browser/engine-gecko/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/components/browser/engine-gecko/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff -r 7a43394d2f4f mobile/android/android-components/components/lib/crash/build.gradle
--- a/mobile/android/android-components/components/lib/crash/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/components/lib/crash/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff -r 7a43394d2f4f mobile/android/android-components/components/service/mars/build.gradle
--- a/mobile/android/android-components/components/service/mars/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/components/service/mars/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff -r 7a43394d2f4f mobile/android/android-components/components/service/nimbus/build.gradle
--- a/mobile/android/android-components/components/service/nimbus/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/components/service/nimbus/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff -r 7a43394d2f4f mobile/android/android-components/components/service/pocket/build.gradle
--- a/mobile/android/android-components/components/service/pocket/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/components/service/pocket/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff -r 7a43394d2f4f mobile/android/android-components/components/support/remotesettings/build.gradle
--- a/mobile/android/android-components/components/support/remotesettings/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/components/support/remotesettings/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -3,14 +3,9 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
     dependencies {
         classpath  ComponentsDependencies.plugin_serialization
diff -r 7a43394d2f4f mobile/android/android-components/plugins/config/build.gradle
--- a/mobile/android/android-components/plugins/config/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/plugins/config/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -7,14 +7,9 @@
 }
 
 repositories {
-    gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-        maven {
-            url repository
-            if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                allowInsecureProtocol = true
-            }
-        }
-    }
+    mavenLocal()
+    maven { url "https://plugins.gradle.org/m2/" }
+    google()
 }
 
 gradlePlugin {
diff -r 7a43394d2f4f mobile/android/android-components/plugins/config/settings.gradle
--- a/mobile/android/android-components/plugins/config/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/plugins/config/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -11,13 +11,8 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 }
diff -r 7a43394d2f4f mobile/android/android-components/plugins/dependencies/build.gradle
--- a/mobile/android/android-components/plugins/dependencies/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/plugins/dependencies/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -7,14 +7,9 @@
 }
 
 repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 }
 
 gradlePlugin {
diff -r 7a43394d2f4f mobile/android/android-components/plugins/dependencies/settings.gradle
--- a/mobile/android/android-components/plugins/dependencies/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/plugins/dependencies/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -11,13 +11,8 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 }
diff -r 7a43394d2f4f mobile/android/android-components/plugins/publicsuffixlist/build.gradle
--- a/mobile/android/android-components/plugins/publicsuffixlist/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/plugins/publicsuffixlist/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -7,14 +7,9 @@
 }
 
 repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 }
 
 dependencies {
diff -r 7a43394d2f4f mobile/android/android-components/plugins/publicsuffixlist/settings.gradle
--- a/mobile/android/android-components/plugins/publicsuffixlist/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/plugins/publicsuffixlist/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -12,14 +12,9 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     includeBuild("../dependencies")
diff -r 7a43394d2f4f mobile/android/android-components/samples/glean/build.gradle
--- a/mobile/android/android-components/samples/glean/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/samples/glean/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 
         dependencies {
             classpath "org.mozilla.telemetry:glean-gradle-plugin:${Versions.mozilla_glean}"
diff -r 7a43394d2f4f mobile/android/android-components/samples/glean/samples-glean-library/build.gradle
--- a/mobile/android/android-components/samples/glean/samples-glean-library/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/samples/glean/samples-glean-library/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -4,14 +4,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 
         dependencies {
             classpath "org.mozilla.telemetry:glean-gradle-plugin:${Versions.mozilla_glean}"
diff -r 7a43394d2f4f mobile/android/android-components/settings.gradle
--- a/mobile/android/android-components/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/android-components/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -10,14 +10,9 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     includeBuild("plugins/dependencies")
diff -r 7a43394d2f4f mobile/android/fenix/app/build.gradle
--- a/mobile/android/fenix/app/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/fenix/app/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -3,14 +3,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff -r 7a43394d2f4f mobile/android/fenix/build.gradle
--- a/mobile/android/fenix/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/fenix/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -10,14 +10,9 @@
 buildscript {
     // This logic is duplicated in the allprojects block: I don't know how to fix that.
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
@@ -51,14 +46,9 @@
 allprojects {
     // This logic is duplicated in the buildscript block: I don't know how to fix that.
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 
         maven {
             url "${gradle.mozconfig.topobjdir}/gradle/maven"
diff -r 7a43394d2f4f mobile/android/fenix/mozilla-lint-rules/build.gradle
--- a/mobile/android/fenix/mozilla-lint-rules/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/fenix/mozilla-lint-rules/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -6,14 +6,9 @@
 apply plugin: 'kotlin'
 
 repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 }
 
 dependencies {
diff -r 7a43394d2f4f mobile/android/fenix/plugins/apksize/build.gradle
--- a/mobile/android/fenix/plugins/apksize/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/fenix/plugins/apksize/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -7,14 +7,9 @@
 }
 
 repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 }
 
 dependencies {
diff -r 7a43394d2f4f mobile/android/fenix/plugins/apksize/settings.gradle
--- a/mobile/android/fenix/plugins/apksize/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/fenix/plugins/apksize/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -11,13 +11,8 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 }
diff -r 7a43394d2f4f mobile/android/fenix/settings.gradle
--- a/mobile/android/fenix/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/fenix/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -10,14 +10,9 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     includeBuild("../android-components/plugins/publicsuffixlist")
diff -r 7a43394d2f4f mobile/android/focus-android/build.gradle
--- a/mobile/android/focus-android/build.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/focus-android/build.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -7,14 +7,9 @@
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
@@ -82,14 +77,9 @@
 
 allprojects {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
 
         maven {
             url "${gradle.mozconfig.topobjdir}/gradle/maven"
diff -r 7a43394d2f4f mobile/android/focus-android/settings.gradle
--- a/mobile/android/focus-android/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/focus-android/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -10,14 +10,9 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     includeBuild("../android-components/plugins/publicsuffixlist")
diff -r 7a43394d2f4f mobile/android/shared-settings.gradle
--- a/mobile/android/shared-settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/mobile/android/shared-settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -12,14 +12,9 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff -r 7a43394d2f4f settings.gradle
--- a/settings.gradle	Tue Mar 25 17:10:31 2025 +0000
+++ b/settings.gradle	Sun Mar 30 11:30:04 2025 +0300
@@ -9,14 +9,9 @@
     }
 
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     includeBuild("${rootProject.projectDir.absolutePath}/mobile/android/android-components/plugins/config")
